File: Oshop.API/Models/Order.cs
using System;
using System.Collections.Generic;

namespace Oshop.API.Models
{
    public class Order
    {
        public string Id { get; set; }
        public DateTime DatePlaced { get; set; }
        public string Status { get; set; }
        public virtual User User { get; set; }
        public string UserId { get; set; }
        public virtual ICollection<ProductItem> ProductItems { get; set; }
        public virtual OrderShipping OrderShipping { get; set; }
        
    }
    
}

File: Oshop-SPA/src/app/_models/order.ts
import { ProductItem } from './productItem';
import { Shipping } from './shipping';

export class Order {
    id: string;
    datePlaced?: Date;
    status: string;
    userId: string;
    productItems: ProductItem[];
    orderShipping: Shipping;
    orderShippingId: string;
}
